detect.go

```diff
@@ -0,0 +1,12 @@
+package ubi8javabuildpack
+
+import (
+	libcnb "github.com/buildpacks/libcnb/v2"
+)
+
+func Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
+	return libcnb.DetectResult{
+		Pass:  true,
+		Plans: []libcnb.BuildPlan{},
+	}, nil
```

Comment:
It would be cleaner if you moved the detection logic here. Then detect doesn't pass every time. Then `Pass` only happens if `version != ""`.

```
	//read the env vars set via the extension.
	versionb, err := os.ReadFile(javaVersionBuilderFile)
	if err != nil {
		return result, err
	}
	helperb, err := os.ReadFile(javaHelpersBuilderFile)
	if err != nil {
		return result, err
	}

	version := strings.TrimSuffix(string(versionb), "\n")
```

Is that possible, or are these files it's reading only available at build time?

Reply:
Indeed.. the files are only available at build. 
During the detect phase, the extension hasn't executed yet, so there are no files to read. 

Lifecycle runs detect, generate, extend, build, and during generate the dockerfile is created by the extension to write the files (only in the builder image), and then during extension they are created when the dockerfile is applied, and then during build they are present as the build phase runs in the context of the extended build image. 

In the future, there may be better options for extension->buildpack communication, It's certainly something I've discussed with Natalie and it sounds like it might be a common enough pattern to be worth adding to a future update of the Buildpacks spec.